### ISSUE TITLE
Data loss when object passed into createNode with string "7up"

### ISSUE THREAD

**Issue by deanc** [gatsbyjs/gatsby#20302]
## Description

I'm using my own plugin to import data from firestore into gatsby. Source code here:
https://github.com/deanc/gatsby-source-firestore/tree/test

When passing in the a certain object to createNode, the `name` field's value when set to `7up` is lost and becomes `null`.
<img width="1042" alt="Screenshot 2019-12-26 at 11 57 23" src="https://user-images.githubusercontent.com/232423/71475158-f4193a00-27d6-11ea-92a2-d57f2b02bffa.png">

### Steps to reproduce

I've tried to boil this down to a test-case. I've tried passing the exact same params into createNode manually in a fresh gatsby, but I can't reproduce it there. I can only reproduce it in my application. Therefore steps to reproduce are very difficult.

I am 100% certain that the field `name` is set in my data to the string `7up`. I have verified this editing the createNode statement in my `node_modules` directly.

The statement:
`createNode(Object.assign({}, nodeData, nodeMeta));`

`nodeData`:
```
{ created: Timestamp { _seconds: 1438014230, _nanoseconds: 0 },
  views: 570,
  slug: '7up',
  active: 1,
  name: '7up',
  type: 'product',
  updated: Timestamp { _seconds: 1438014230, _nanoseconds: 0 },
  id: 50,
  nameTwo: '7up' }
```

`nodeMeta`:
```
{ id: 'bh0UWRLmzZkmHAJUwQOR',
  parent: null,
  children: [],
  internal:
   { type: 'Item',
     content:
      '{"created":{"_seconds":1438014230,"_nanoseconds":0},"views":570,"slug":"7up","active":1,"name":"7up","type":"product","updated":{"_seconds":1438014230,"_nanoseconds":0},"id":50,"nameTwo":"7up"}',
     contentDigest: '663b0f201d29ac315961f9b111612f33' } }
```

When passing the above params manually in a fresh gatsby install, it works. Tried 3 times in a row.

### Expected result

The `allItem` graphql object should have keys `name` and `slug` with values `7up`.

### Actual result

Instead 9/10 times when running a full cache clear and `yarn start` the `name` field's value is `null`. Very occasionally the `slug` is `null` too.

### Environment

```
  System:
    OS: macOS 10.15.2
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Shell: 5.0.11 - /usr/local/bin/bash
  Binaries:
    Node: 10.16.3 - /usr/local/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 79.0.3945.88
    Firefox: 70.0
    Safari: 13.0.4
  npmPackages:
    gatsby: ^2.18.8 => 2.18.8
    gatsby-firesource: https://github.com/deanc/gatsby-source-firestore#test => 2.0.0
    gatsby-image: ^2.2.34 => 2.2.34
    gatsby-plugin-layout: ^1.1.18 => 1.1.18
    gatsby-plugin-manifest: ^2.2.31 => 2.2.31
    gatsby-plugin-offline: ^3.0.27 => 3.0.27
    gatsby-plugin-react-helmet: ^3.1.16 => 3.1.16
    gatsby-plugin-sass: ^2.1.26 => 2.1.26
    gatsby-plugin-sharp: ^2.3.5 => 2.3.5
    gatsby-plugin-styled-components: ^3.1.16 => 3.1.16
    gatsby-source-filesystem: ^2.1.43 => 2.1.43
    gatsby-transformer-sharp: ^2.3.7 => 2.3.7
  npmGlobalPackages:
    gatsby-cli: 2.8.22
```

### Conclusion

I have 70 other items in my firestore database. This only happens when the name field is set to `7up`. I just changed it to `6up` and ran the `rm -rf .cache/; yarn start` command 5 times, and the `name` field was correctly populated.

My guesses are either this is deep in the depths of gatsby there is some transformation happening on that string and somehow it's thinking it's some form of number representation. Or, there is some kind of clash between this and existing records. 

All ideas welcome, and happy to debug further.

**Comment by vladar** [comment]
This sounds very weird. It is possible that some of your other nodes affect this one. So maybe try to reproduce it locally with a full dataset? I see you fetch nodes in parallel, so maybe there is some kind of race condition here (which would explain why it happens 9 times out of 10).

Also, do you have any warnings in the console? We definitely need a reproduction to continue this investigation.

**Comment by deanc** [comment]
There's no warnings in the console at all. I can't see how the other nodes *could* affect it. I'll try and reduce to a test-case. I understand it's tricky without it. Was kind of hoping someone had some instincts to help me try and debug further.

**Comment by deanc** [comment]
I've managed to reduce this to a test-case now.

1.  Check out this repo: https://github.com/deanc/gatsby-test (I moved the plugin logic into gatsby-node to avoid any additional dependencies) and run `yarn install`
2. Download your credentials.json file from Firestore admin and place it in the root of the project with name `credentials.json`.
3. Create a new collection called `items2` and give it this exact data:
<img width="406" alt="Screenshot 2019-12-26 at 17 35 33" src="https://user-images.githubusercontent.com/232423/71484828-49207480-2806-11ea-8726-69c1b30125de.png">
4. Run `yarn start` and then browse to the graphql viewer.

# What i expected to happen

My fields would be populated and all there.

# What actually happens

Fields are actually not present, and also sometimes `null`.

**Comment by vladar** [comment]
Looks like there was a bug in `gatsby 2.18.12`. In short, `sanitizeNode` was removing keys with identical values. This bug was fixed in `gatsby 2.18.13` via https://github.com/gatsbyjs/gatsby/pull/20039

Please try upgrading.

**Comment by deanc** [comment]
@vladar that was the issue. Thanks. I had tried upgrading gatsby-cli earlier, but didn't realise my dependencies were out of date.
